osu.Game/Skinning/LegacyRankDisplay.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable disable
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Rulesets.Scoring;
+
+namespace osu.Game.Skinning
+{
+    public partial class LegacyRankDisplay : CompositeDrawable, ISerialisableDrawable
+    {
+        public bool UsesFixedAnchor { get; set; }
+
+        [Resolved]
+        private ScoreProcessor scoreProcessor { get; set; } = null!;
+
+        [Resolved]
+        private ISkinSource source { get; set; } = null!;
+
+        private readonly Sprite rank;
+
+        public LegacyRankDisplay()
+        {
+            AutoSizeAxes = Axes.Both;
+
+            AddInternal(rank = new Sprite());
+        }
+
+        protected override void LoadComplete()
+        {
+            var skin = source.FindProvider(s => getTexture(s, "A") != null);
```

Comment:
Why does this use `FindProvider()` at all rather than just getting the texture directly from the skin? Why the A rank specifically? I don't see any checks of this sort in stable.

Also, doesn't the fact that this doesn't respond to `source.SkinChanged` at all mean that this implementation just does not support changing skin mid-gameplay?

Reply:
How do I change skin mid-gameplay? Do I need to create a visual test for this?

Reply:
- use "Random skin" hotkey (<kbd>Ctrl</kbd>-<kbd>Shift</kbd>-<kbd>R</kbd> by default)
- play a replay and change via settings sidebar (<kbd>Ctrl</kbd>-<kbd>O</kbd>)
- in normal gameplay. pause, open settings sidebar, change skin

Reply:
oh it does change